packages/solana/src/connectors/walletAdapters.ts

```diff
@@ -0,0 +1,39 @@
+import { PhantomWalletAdapter } from '@solana/wallet-adapter-phantom'
+import { SolflareWalletAdapter } from '@solana/wallet-adapter-solflare'
+import { TrustWalletAdapter } from '@solana/wallet-adapter-trust'
+import { BackpackWalletAdapter } from '@solana/wallet-adapter-backpack'
+
+import type { BaseWalletAdapter } from '@solana/wallet-adapter-base'
+import type { Connector } from '@web3modal/scaffold'
+
+export type AdapterKey = 'phantom' | 'solflare' | 'trustWallet' | 'backpack'
+export const supportedWallets: AdapterKey[] = ['phantom', 'solflare', 'trustWallet', 'backpack']
+
+export function createWalletAdapters() {
+  return {
+    phantom: new PhantomWalletAdapter(),
+    trustWallet: new TrustWalletAdapter(),
+    backpack: new BackpackWalletAdapter(),
+    solflare: new SolflareWalletAdapter()
+  }
+}
+
+export function syncInjectedWallets(
+  w3mConnectors: Connector[],
+  adapters: Record<AdapterKey, BaseWalletAdapter>
+) {
+  supportedWallets.forEach(wallet => {
+    if (window[wallet as keyof Window]) {
+      w3mConnectors.push({
+        id: adapters[wallet].name,
+        type: 'ANNOUNCED',
+        imageUrl: adapters[wallet].icon,
+        name: adapters[wallet].name,
+        provider: adapters[wallet],
+        info: {
+          rdns: adapters[wallet].publicKey?.toString()
```

Comment:
hm, rdns is tied to the provider rather than the address itself. 
Technically should be a name for which you can do a reverse lookup for info on it but if it doesn't really exist in solana 🤷 

I'd use the wallet name or similar instead of the address though, since it's more inline with evm providers.

cc @glitch-txs what do you think?